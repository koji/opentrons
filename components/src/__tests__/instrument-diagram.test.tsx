import React from 'react'
import Renderer from 'react-test-renderer'

import { InstrumentDiagram, InstrumentGroup } from '..'

// TODO(bc, 2021-03-03): unit test this component and split out InstrumentDiagram from InstrumentGroup

describe('InstrumentDiagram', () => {
  it('Single-channel renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram
        mount="left"
        pipetteSpecs={{ channels: 1, displayCategory: 'GEN1' }}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Multi-channel renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram
        mount="left"
        pipetteSpecs={{ channels: 8, displayCategory: 'GEN1' }}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Single-channel GEN2 renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram
        mount="right"
        pipetteSpecs={{ channels: 1, displayCategory: 'GEN2' }}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Multi-channel GEN2 renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram
        mount="right"
        pipetteSpecs={{ channels: 8, displayCategory: 'GEN2' }}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('single-channel FLEX renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram
        mount="right"
        pipetteSpecs={{ channels: 1, displayCategory: 'FLEX' }}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('eight-channel FLEX renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram
        mount="right"
        pipetteSpecs={{ channels: 8, displayCategory: 'FLEX' }}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('96-channel GEN1 renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram
        mount="left"
        pipetteSpecs={{ channels: 96, displayCategory: 'GEN1' }}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('InstrumentGroup', () => {
  it('Renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentGroup
        left={{
          mount: 'left',
          description: 'p300 8-Channel',
          pipetteSpecs: { channels: 8, displayCategory: 'GEN1' },
          isDisabled: false,
          className: 'foo',
        }}
        right={{
          mount: 'right',
          description: 'p10 Single',
          pipetteSpecs: { channels: 1, displayCategory: 'GEN1' },
          isDisabled: true,
          className: 'blah',
        }}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
