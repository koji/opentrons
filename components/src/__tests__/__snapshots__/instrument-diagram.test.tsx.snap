// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstrumentDiagram 96-channel GEN1 renders correctly 1`] = `
.c0 {
  min-width: 0;
}

<div
  className="c0"
>
  <img
    className=""
    src="ninety-six-channel-gen1.png"
  />
</div>
`;

exports[`InstrumentDiagram Multi-channel GEN2 renders correctly 1`] = `
.c0 {
  min-width: 0;
}

<div
  className="c0"
>
  <img
    className="flipped_image"
    src="multi-channel_GEN2_800px.png"
  />
</div>
`;

exports[`InstrumentDiagram Multi-channel renders correctly 1`] = `
.c0 {
  min-width: 0;
}

<div
  className="c0"
>
  <img
    className=""
    src="multi-channel_GEN1_800px.png"
  />
</div>
`;

exports[`InstrumentDiagram Single-channel GEN2 renders correctly 1`] = `
.c0 {
  min-width: 0;
}

<div
  className="c0"
>
  <img
    className="flipped_image"
    src="single-channel_GEN2_800px.png"
  />
</div>
`;

exports[`InstrumentDiagram Single-channel renders correctly 1`] = `
.c0 {
  min-width: 0;
}

<div
  className="c0"
>
  <img
    className=""
    src="single_channel_GEN1_800px.png"
  />
</div>
`;

exports[`InstrumentDiagram eight-channel GEN3 renders correctly 1`] = `
.c0 {
  min-width: 0;
}

<div
  className="c0"
>
  <img
    className="flipped_image"
    src="eight-channel-gen3.png"
  />
</div>
`;

exports[`InstrumentDiagram single-channel GEN3 renders correctly 1`] = `
.c0 {
  min-width: 0;
}

<div
  className="c0"
>
  <img
    className="flipped_image"
    src="single-channel-gen3.png"
  />
</div>
`;

exports[`InstrumentGroup Renders correctly 1`] = `
.c0 {
  min-width: 0;
}

<section
  className="pipette_group"
>
  <div
    className="pipette left foo"
  >
    <div
      className="pipette_info"
    >
      <div>
        <h2
          className="title"
        >
          pipette
        </h2>
        <span
          className="value"
        >
          p300 8-Channel
        </span>
      </div>
    </div>
    <div
      className="c0 pipette_icon"
    >
      <img
        className=""
        src="multi-channel_GEN1_800px.png"
      />
    </div>
  </div>
  <div
    className="pipette right disabled blah"
  >
    <div
      className="pipette_info"
    >
      <div>
        <h2
          className="title"
        >
          pipette
        </h2>
        <span
          className="value"
        >
          p10 Single
        </span>
      </div>
    </div>
    <div
      className="c0 pipette_icon"
    >
      <img
        className="flipped_image"
        src="single_channel_GEN1_800px.png"
      />
    </div>
  </div>
</section>
`;
